### PR TITLE
Add version info to common package and -version flag to CLI

### PR DESCRIPTION
**Pull request by tjanez**

### CODE REVIEW

File: common/version.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+var (
+	// SoftwareVersion represents the Oasis Core Rosetta Gateway's version and
+	// should be set by the linker.
+	SoftwareVersion = "0.0.0-unset"
+
+	// RosettaAPIVersion represents the Rosetta API version with which the
+	// Oasis Core Rosetta Gateway is guaranteed to be compatible with.
+	RosettaAPIVersion = "1.4.1"
```
Comment (tjanez) [inline common/version.go:16]: This shouldn't be hard-coded in the future, see: #129.

File: main.go [hunk 2]
```diff
@@ -113,29 +119,50 @@ func getEnvVarOrExit(name string) string {
...
 		os.Exit(1)
 	}
+	return port
+}
+
+// Print version information.
+func printVersionInfo() {
+	fmt.Printf("Software version: %s\n", common.SoftwareVersion)
+	fmt.Printf("Rosetta API version: %s\n", common.RosettaAPIVersion)
+	fmt.Printf("Go toolchain version: %s\n", common.ToolchainVersion)
```
Comment (kostko) [inline main.go:143]: Minor consistency nit, in Oasis Core we align (pad with spaces) all the versions in the same group.

**Review by kostko — APPROVED** [review]
Cool! Maybe it would be a good idea to also include the Oasis Core version? Well it would actually make more sense to include the client protocol version once oasisprotocol/oasis-core#1861 is done.

File: common/version.go (hunk 1 above)
Comment (abukosek) [reply common/version.go:16]: You should also make https://github.com/oasisprotocol/oasis-core-rosetta-gateway/blob/master/services/network.go#L135 use the RosettaAPIVersion specified here :)

Comment (tjanez) [reply common/version.go:16]: > You should also make https://github.com/oasisprotocol/oasis-core-rosetta-gateway/blob/master/services/network.go#L135 use the RosettaAPIVersion specified here :)

Good catch, fixed.

**Review by abukosek — APPROVED** [review]

**Comment by tjanez** [comment]
> Maybe it would be a good idea to also include the Oasis Core version? Well it would actually make more sense to include the client protocol version once [oasisprotocol/oasis-core#1861](https://github.com/oasisprotocol/oasis-core/issues/1861) is done.

Agreed with adding the client protocol version once that is implemented in Oasis Core.

I tried adding the Oasis Core version by obtaining its value from the [`common.version.SoftwareVersion` variable](https://pkg.go.dev/github.com/oasisprotocol/oasis-core/go@v0.2012.2/common/version#pkg-variables), but that will return `0.0-unset` since that variable is only set through a linker flag when compiling...

File: main.go (hunk 2 above)
Comment (tjanez) [reply main.go:143]: If there's an elegant way to get the Oasis Core version from `go.mod` (or similar), we could have something like:

```
Software version: 1.0.0
Oasis Core:
  Software version: 20.12.2
Rosetta API version: 1.4.1
Go toolchain version: 1.15.5
```
and later:
```
Software version: 1.0.0
Oasis Core:
  Software version:        20.12.2
  Client protocol version: 1.0.0
Rosetta API version: 1.4.1
Go toolchain version: 1.15.5
```

Or did you mean we should expose Oasis Core's consensus protocol versions here?

As for aligning the Software, Rosetta API and Go toolchain versions, I don't think we align them in Oasis Node either?

Comment (kostko) [reply main.go:143]: > Or did you mean we should expose Oasis Core's consensus protocol versions here?

No, just the software version.

> As for aligning the Software, Rosetta API and Go toolchain versions, I don't think we align them in Oasis Node either?

Well we align in the same group as above (software and go toolchain versions are each in their own group in oasis-node).

Comment (tjanez) [reply main.go:143]: > > Or did you mean we should expose Oasis Core's consensus protocol versions here?
> 
> No, just the software version.

Ok, cool.

> > As for aligning the Software, Rosetta API and Go toolchain versions, I don't think we align them in Oasis Node either?
> 
> Well we align in the same group as above (software and go toolchain versions are each in their own group in oasis-node).

Aha, ok. I would prefer treating software, Rosetta API and Go toolchain as separate groups, similar to Oasis Node and other Oasis Core CLI tools.

Comment (kostko) [reply main.go:143]: Sounds good.

**Comment by tjanez** [comment]
> Cool! Maybe it would be a good idea to also include the Oasis Core version?

Implemented in #134.

> Well it would actually make more sense to include the client protocol version once [oasisprotocol/oasis-core#1861](https://github.com/oasisprotocol/oasis-core/issues/1861) is done.

Filled #135.

**Pull request merged**